config-schema.json

```diff
@@ -0,0 +1,127 @@
+{
```

Comment:
@craigpastro @miparnisari do you think we should try and autogenerate our default Config struct based on the "default" values in this `config-schema.json` file?

It would make this the source of truth, which would be good for documentation, but it would be a bit more complicated of an integration in this project if we did so.

Personally, I don't mind having to update the `config-schema.json` file along with the changes to the default values (if we make changes to the defaults). At least this now lives alongside the source code, and we can generate doc in openfga.dev off of this file.

Reply:
I was playing around with something yesterday and found that the grcp backoff package provides a [DefaultConfig](https://pkg.go.dev/google.golang.org/grpc/backoff#pkg-variables), and I thought that was a very cool way to specify a default. Would something like that make integration easier? Or we could even make a `config` package and a `func Default() *Config` function or something like that?

I feel something like this may easily get out of date.

Reply:
Yeah, the `func Default() *Config` would serve the same purpose that the `GetServiceConfig()` func today is serving.

The purpose of the `config-schema.json` is so that our documentation and other sources can generate doc based on the schema. If we wanted to show a default value column in a table we'd want to use the "default" keyword in our json-schema wherever it is applicable to a config with a default value. As you pointed out though, it would be easy for the defaults in `config-schema.json` to get out of date with the actual defaults encoded in the `GetServerConfig()` function. This conundrum is the basis of my question.

We could, within the `GetServiceConfig` function, use the `config-schema.json` to generate the default values. That would be possible, but it's also a little overkill if you ask me, because we'd have to include the json file in our containers and we'd have to map a json-schema to a config struct in Go etc.. It just seems like a lot for a little benefit 🤷